src/toolkit/metadatas.cc

```diff
@@ -0,0 +1,223 @@
+#include <math.h>
+#include <stdio.h>
+#include <sstream>
+
+#include "agent.h"
+#include "context.h"
+#include "metadatas.h"
+namespace cyclus {
+namespace toolkit {
+
+enum type_hash {
+  s,  // string
+  b,  // boolean
+  i,  // int
+  u,  // uint
+  d   // double
+};
+
+std::unordered_map<std::string, int> type_map = {
+    std::make_pair("s", s),  // string
+    std::make_pair("b", b),  // boolean
+    std::make_pair("i", i),  // int
+    std::make_pair("u", u),  // uint
+    std::make_pair("d", d)   // double
+};
+
+Metadatas::Metadatas() {}
+Metadatas::~Metadatas() {}
+
+void Metadatas::RecordMetadatas(Agent* agent) {
+  std::string tblname = "Metadatas";
+  Json::Value::Members keys = metadatas.getMemberNames();
+  Json::Value::Members::const_iterator ikey = keys.begin();
+  Json::Value::Members::const_iterator ikey_end = keys.end();
+  for (; ikey != ikey_end; ++ikey) {
+    std::string value = "";
+    std::string type = "";
+    switch (metadatas[*ikey].type()) {
+      case Json::nullValue:
+        break;
+      case Json::stringValue:
+        type = "string";
+        value = metadatas[*ikey].asString();
+        break;
+      case Json::booleanValue:
+        type = "bool";
+        value = metadatas[*ikey].asString();
+        break;
+      case Json::intValue:
+        type = "int";
+        value = std::to_string(metadatas[*ikey].asInt());
+        break;
+      case Json::uintValue:
+        type = "uint";
+        value = std::to_string(metadatas[*ikey].asInt());
+        break;
+      case Json::realValue:
+        type = "double";
+        value = std::to_string(metadatas[*ikey].asDouble());
+        break;
+      case Json::arrayValue:
+      case Json::objectValue:
+        ValueError("Type is not convertible to string");
+      default:
+        ValueError("Type is not known.");
+    }
+    if (value != "") {
```

Comment:
I think that `value != ""` will evaluate as true even if value is Null. Maybe worth a test?